pkg/api/methods.go

```diff
@@ -44,3 +49,47 @@ func (t *Taint) ToString() string {
 	}
 	return fmt.Sprintf("%v=%v:%v", t.Key, t.Value, t.Effect)
 }
+
+func (obj *ObjectReference) SetGroupVersionKind(gvk schema.GroupVersionKind) {
+	obj.APIVersion, obj.Kind = gvk.ToAPIVersionAndKind()
+}
+func (obj *ObjectReference) GroupVersionKind() schema.GroupVersionKind {
+	return schema.FromAPIVersionAndKind(obj.APIVersion, obj.Kind)
+}
+
+func (obj *ObjectReference) GetObjectKind() schema.ObjectKind { return obj }
+
+// Returns string version of ResourceName.
+func (self ResourceName) String() string {
```

Comment:
same as comment on the other PR, but I'd rather these go in a resource-specific file?